x/estaking/modules/distribution/module.go

```diff
@@ -178,3 +181,88 @@ func (am AppModule) AllocateTokens(ctx sdk.Context) {
 		panic(err)
 	}
 }
+
+// AllocateTokens handles distribution of the collected fees
+// Handles EdenB rewards
+func (am AppModule) AllocateEdenBTokens(ctx sdk.Context) {
+	// fetch and clear the collected fees for distribution, since this is
+	// called in BeginBlock, collected fees will be from the previous block
+	// (and distributed to the current representatives)
+	feeCollector := am.accountKeeper.GetModuleAccount(ctx, am.feeCollectorName)
+	feesCollectedInt := am.bankKeeper.GetAllBalances(ctx, feeCollector.GetAddress())
+
+	filteredCoins := FilterDenoms(feesCollectedInt, ptypes.EdenB)
+	feesCollected := sdk.NewDecCoinsFromCoins(filteredCoins...)
+
+	// transfer collected fees to the distribution module account
+	if filteredCoins.IsAllPositive() {
+		err := am.bankKeeper.SendCoinsFromModuleToModule(ctx, am.feeCollectorName, distrtypes.ModuleName, filteredCoins)
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	// calculate the fraction allocated to representatives by subtracting the community tax.
+	// e.g. if community tax is 0.02, representatives fraction will be 0.98 (2% goes to the community pool and the rest to the representatives)
+	remaining := feesCollected
+	communityTax, err := am.keeper.GetCommunityTax(ctx)
+	if err != nil {
+		panic(err)
+	}
+	representativesFraction := math.LegacyOneDec().Sub(communityTax)
+
+	edenBValidator := am.estakingKeeper.GetEdenBValidator(ctx).GetOperator()
+
+	// Note: to prevent negative coin amount issue when invariant's broken,
+	// calculation of total bonded tokens manually through iteration
+	sumOfValTokens := math.ZeroInt()
+	err = am.estakingKeeper.IterateBondedValidatorsByPower(ctx, func(_ int64, validator stakingtypes.ValidatorI) bool {
```

Comment:
Let's add more details. like EdenB is distributed for staking Elys and locking Eden, not for locking EdenB